diff --git test/test-connect-unspecified.c test/test-connect-unspecified.c
index 04e1c8a5..5f32b67a 100644
--- test/test-connect-unspecified.c
+++ test/test-connect-unspecified.c
@@ -48,12 +48,14 @@ TEST_IMPL(connect_unspecified) {
                         (const struct sockaddr*) &addr4,
                         connect_4) == 0);
 
-  ASSERT(uv_tcp_init(loop, &socket6) == 0);
-  ASSERT(uv_ip6_addr("::", TEST_PORT, &addr6) == 0);
-  ASSERT(uv_tcp_connect(&connect6,
-                        &socket6,
-                        (const struct sockaddr*) &addr6,
-                        connect_6) == 0);
+  if (can_ipv6()) {
+    ASSERT(uv_tcp_init(loop, &socket6) == 0);
+    ASSERT(uv_ip6_addr("::", TEST_PORT, &addr6) == 0);
+    ASSERT(uv_tcp_connect(&connect6,
+                          &socket6,
+                          (const struct sockaddr*) &addr6,
+                          connect_6) == 0);
+  }
